module/support/ReactionStats/src/ReactionStats.cpp

```diff
@@ -0,0 +1,46 @@
+#include "ReactionStats.hpp"
+
+#include "extension/Configuration.hpp"
+
+#include "message/support/nuclear/ReactionStatistics.hpp"
+
+namespace module::support {
+
+    using extension::Configuration;
+    using NUClear::message::ReactionStatistics;
+    using ReactionStatisticsProto = message::support::nuclear::ReactionStatistics;
+
+    ReactionStats::ReactionStats(std::unique_ptr<NUClear::Environment> environment) : Reactor(std::move(environment)) {
+
+        on<Configuration>("ReactionStats.yaml").then([this](const Configuration& config) {
+            // Use configuration here from file ReactionStats.yaml
+        });
+
+        on<Trigger<ReactionStatistics>>().then([this](const ReactionStatistics& stats) {
+            auto reactionData               = std::make_unique<ReactionStatisticsProto>();
+            reactionData->name              = stats.identifier[0];
+            reactionData->trigger_name      = stats.identifier[1];
+            reactionData->function_name     = stats.identifier[2];
+            reactionData->reaction_id       = stats.reaction_id;
+            reactionData->task_id           = stats.task_id;
+            reactionData->cause_reaction_id = stats.cause_reaction_id;
+            reactionData->cause_task_id     = stats.cause_task_id;
+            reactionData->emitted           = 0;
+            reactionData->started           = 0;
+            reactionData->finished          = 0;
+            // TODO fix this
+            /*reactionData.emitted           = getUtcTimestamp<std::chrono::microseconds>(stats.emitted);
+            reactionData.started           = getUtcTimestamp<std::chrono::microseconds>(stats.started);
+            reactionData.finished          = getUtcTimestamp<std::chrono::microseconds>(stats.finished);*/
+
+            emit(reactionData);
+
+
+            log(stats.identifier[0],
+                1000.0
+                    * (double((stats.finished - stats.started).count())
+                       / double(NUClear::clock::duration::period::den)),
+                "ms");
```

Comment:
If you're doing math with times, just use the chrono tools. In this case it looks like you're after the time in milliseconds

`std::chrono::duration_cast<std::chrono::milliseconds(stats.finished - stats.started).count()`

If you really wanted it as a double, might be good to do this
`std::chrono::duration_cast<std::chrono::duration<double>>(stats.finished - stats.started).count() * 1000`

Reply:
Gotcha, this was just leftover copypasta from the file I used as a template, Once I've finished testing I'll be removing that line anyway but I'll do it for now while I verify 👍